### PR TITLE
Feat/add instance location

### PR DESCRIPTION
**Pull request by dcuenot-sfdc**
**Problem to solve**
Display Location and Release of Instances directly in the frontend of this application, in order to easily create Push Upgrades groups. More informations in this [Quip Doc](https://salesforce.quip.com/OJypA4wf657K)

**A way to fix that issue**
To get some informations on all public instances, we can do an API call to api.status.salesforce.com.
Based on this info, I created a worker to update informations in the database, and display some info on the admin panel.

**How what it tested?**
No unit tests, but I ran the application on my laptop with a snapshot of the staging Database.

### CODE REVIEW

File: api/orgs.js [hunk 1]
```diff
@@ -21,6 +21,9 @@ const QUERY_DICTIONARY = new Map([
 	["edition", "o.edition"],
 	["instance", "o.instance"],
 	["type", "o.type"],
+	["location", "o.org_location"],
+	["env", "o.org_env"],
+	["release", "o.org_release"],
```
Comment (michaelhoefer) [inline api/orgs.js:26]: org_xxx is not I think the right name here.  The table is org.  org.org_xxx is redundant.  org_location can probably be location, org_release can be release or release_schedule.  org_env can just be environment.

File: init/sql/migrate.sql [hunk 2]
```diff
@@ -15,3 +15,9 @@ alter table package_version
 alter table package_version_latest
     alter column version_sort type varchar(30),
     alter column limited_version_sort type varchar(30);
+
+alter table org
+    add if not exists org_location varchar(30),
+    add if not exists org_env varchar(30),
+    add if not exists org_release varchar(30),
+    add if not exists org_status varchar(30);
```
Comment (michaelhoefer) [inline init/sql/migrate.sql:23]: This should just be status,  but we already have a status.  So this needs to either use the existing status field, or use a more appropriate name that distinguishes it from our current status field, determined by what value is actually represents.

File: worker/orgfetch.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 const db = require('../util/pghelper');
 const opvapi = require('../api/orgpackageversions');
 const orgsapi = require('../api/orgs');
+const request = require('request');
+const { logger } = require('../util/logger');
+
+const INTERNAL_ORGS = ['GS0', 'CS46', 'CS49'];
```
Comment (michaelhoefer) [inline worker/orgfetch.js:7]: These are instances, not orgs.  And we should make them a config variable with these three as default values.  e.g. `const INTERNAL_INSTANCES = JSON.parse(process.env.INTERNAL_INSTANCES || ['GS0', 'CS46', 'CS49']);
`

File: worker/orgfetch.js [hunk 4]
```diff
@@ -31,6 +35,57 @@ async function updateChildrenFromParents(job) {
 	}
 }
 
+async function updateOrgLocation(job) {
+	adminJob = job;
+
+	try {
+		let query = ` SELECT DISTINCT ON (instance) instance AS key, org_location, org_env, org_release, org_status FROM org `;
+		let instances_from_db = await db.query(query);
```
Comment (michaelhoefer) [inline worker/orgfetch.js:43]: Code style => no underscores in variable names.  underscores only appropriate for constants w/ all caps.

**Review by michaelhoefer — CHANGES_REQUESTED** [review]
First off, great job figuring out the codebase to contribute this, really appreciate it.

Second, after getting to the end of the changes, I think you should take a different approach.  You are trying to load all instances on every sync operation, and update all orgs if something changed, effectively stuffing instance data into the orgs table.  I suggested name changes below, but even those are not really correct.  It isn't org_env, or env....it is instance_env.

Rather, it should be instance.env, instance.location, instance.release_schedule, etc.  meaning, you should create a new table in the database called instances, store the instance data from status.salesforce.com locally, and join the instance data to the org data where appropriate.  Like how we handle accounts.

You will still have to I think query all instances from the rest api each time, because I don't think it gives you a select-since-last-modified capability like we get with our SOQL queries.  But keeping it just an exercise in one-way syncing the instance data to the local db, and then letting you use that instance data however you see fit, is a better, cleaner, simpler implementation.

**Comment by dcuenot-sfdc** [comment]
Thanks @michaelhoefer for your feedbacks.
Indeed I did a mix between orgs and instances => I will fix that.

> you should create a new table in the database called instances, store the instance data from status.salesforce.com locally, and join the instance data to the org data where appropriate.

To be honest, it was my first idea, but when I saw 1651 instances in the status.salesforce.com answer, and only 100 different instances in package-manager db, I changed my mind. But if I step back a little, I think that the right approach, so I will change this also.

**Review by michaelhoefer — APPROVED** [review]
Looks good to me.  Pls review the couple of changes I made.

**Comment by dcuenot-sfdc** [comment]
Thanks for the review.
All changes look good to me.

Should I do something else before merging into main?

**Comment by michaelhoefer** [comment]
Nice job Damien.

**Pull request merged**